### PR TITLE
[INFINITY-3256] MIT KDC swap in place of Heimdal

### PR DESCRIPTION
**Pull request by NimaVaziri**
This PR replaces the Heimdal implementation of the KDC with the MIT implementation.

- [X] Replace KDC server implementation
- [ ] Replace client side implementation

### CODE REVIEW

File: tools/kdc/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+    apt-get clean
+
+
+# The following Kerberos distribution is derived from
+# https://kerberos.org/dist/krb5/1.15/krb5-1.15.2.tar.gz
+RUN wget https://s3-us-west-2.amazonaws.com/nima-dev/krb5-1.16.tar.gz
+RUN tar xvf krb5-1.16.tar.gz  && rm krb5-1.16.tar.gz
+RUN cd krb5-1.16/src && ./configure && make && make install
+
+RUN mkdir /kdc && \
```
Comment (NimaVaziri) [inline tools/kdc/Dockerfile]: Still configuring this, need to enable functionality in strict.

File: tools/kdc/krb5.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[logging]
+ default = FILE:/var/log/krb5libs.log
+ kdc = FILE:/var/log/krb5kdc.log
+ admin_server = FILE:/var/log/kadmind.log
+
+[libdefaults]
+ default_realm = LOCAL
```
Comment (elezar) [inline tools/kdc/krb5.conf]: Should this not refer to `REALM` as well?

File: tools/kdc/setup_kdc.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# setup the database used by the KDC
+/setup_kdc_db.sh
+
+sed -i "s/PORT_KDC/$PORT_KDC/g" /kdc/kdc.conf
```
Comment (elezar) [inline tools/kdc/setup_kdc.sh]: How about using mustache for the templating instead of bash?

File: tools/kdc/setup_kdc.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
```
Comment (elezar) [inline tools/kdc/setup_kdc.sh]: Should we add a `set -e` here so that this fails fast?

File: tools/kdc/setup_kdc.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+# setup the database used by the KDC
+/setup_kdc_db.sh
+
+sed -i "s/PORT_KDC/$PORT_KDC/g" /kdc/kdc.conf
+sed -i "s/PORT_KDC/$PORT_KDC/g; s/REALM/$REALM/g" /kdc/krb5.conf
+krb5kdc
```
Comment (elezar) [inline tools/kdc/setup_kdc.sh]: Does this start in background mode?

File: tools/kdc/setup_kdc_db.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/expect
+
+spawn /usr/local/sbin/kdb5_util create -s -r $env(REALM)
```
Comment (elezar) [inline tools/kdc/setup_kdc_db.sh]: According to https://linux.die.net/man/8/kdb5_util one could use the following option:
```
-P password
specifies the master database password. This option is not recommended.
```

File: tools/kdc/Dockerfile [hunk 7]
```diff
@@ -1,11 +1,40 @@
...
+      curl \
+      expect \
+      less \
+      vim \
+      wget && \
+    apt-get clean
+
+
+# The following Kerberos distribution is derived from
+# https://kerberos.org/dist/krb5/1.15/krb5-1.15.2.tar.gz
```
Comment (elezar) [inline tools/kdc/Dockerfile]: What is the reason for the version change in your tar.gz file?

**Comment by NimaVaziri** [comment]
The switch doesn't add any benefits as we previously thought it would.

**Pull request closed without merging**